mobile/src/screens/BookScreen.tsx

```diff
@@ -0,0 +1,185 @@
+import { I18nContext } from '../context/I18nContext';
+import { TextStyles } from '../styles/TextStyles';
+import { ButtonGroup } from '../components/ButtonGroup';
+import { LanguageButtons } from '../components/LanguageButtons';
+
+/**
+ * Individual book view displaying book details
+ */
+export const BookScreen: React.FC = () => {
+  // get book id and langs from route params
+  const route = useRoute<BookScreenRouteProps>();
+  const { id, langs } = route.params;
+
+  const client = useContext(APIContext);
+  const i18nCtx = useContext(I18nContext);
+
+  // book screen states
+  const [loading, setLoading] = useState(true);
+  const [language, setLanguage] = useState(i18nCtx.locale.substring(0, 2));
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [image, setImage] = useState('');
+  const [tabBody, setTabBody] = useState('');
+  const [tabVideo, setTabVideo] = useState('');
+  const [activeButton, setActiveButton] = useState('');
+  const [tabContent, setTabContent] = useState(null);
+
+  const markdownStyles = {
+    heading: { fontFamily: 'Gotham-Light' },
+    paragraph: { fontFamily: 'Gotham-Light' },
+    strong: { fontFamily: 'Gotham-Bold' },
+    listItemNumber: { fontFamily: 'Gotham-Light' },
+    listItemOrderedContent: { fontFamily: 'Gotham-Light' },
+    listItemUnorderedContent: { fontFamily: 'Gotham-Light' },
+    tableHeaderCellContent: { fontFamily: 'Gotham-Light' },
+    em: { fontFamily: 'Gotham-Italic' },
+  };
+
+  // fetches book details on language change
+  useEffect(
+    () => {
+      setLoading(true);
+      let lang;
+
+      if (langs.indexOf(language) > -1) {
+        lang = language;
+      } else if (langs.indexOf('en') > -1) {
+        lang = 'en';
+        setLanguage(lang);
+      } else {
+        const [first] = langs;
+        lang = first;
+        setLanguage(lang);
+      }
+
+      client.getBook(id, lang).then((res) => {
+        setTitle(res.title);
+        setAuthor(res.author);
+        setImage(res.image);
+        setActiveButton('btn-1');
+        setTabContent({
+          'btn-1': res.read,
+          'btn-2': res.explore,
+          'btn-3': res.learn,
+        });
+        setLoading(false);
+      }).catch((err) => {
+        console.log(err);
+      });
+    },
+    [language],
+  );
+
+  // sets tab content on tab button change
+  useEffect(
+    () => {
+      if (tabContent) {
+        setTabVideo(null);
+        if (tabContent[activeButton].video) {
+          const regEx = '^(?:https?:)?//[^/]*(?:youtube(?:-nocookie)?.com|youtu.be).*[=/]([-\\w]{11})(?:\\?|=|&|$)';
+          const matches = tabContent[activeButton].video.match(regEx);
+          if (matches) {
+            setTabVideo(matches[1]);
+          }
+        }
+        setTabBody(tabContent[activeButton].body);
+      }
+    },
+    [activeButton, tabContent],
+  );
+
+  return (
+    loading ? <View style={styles.loadingCircle}><LoadingCircle /></View> :
```

Comment:
nit: it might be better to just

```ts
if (loading) {
  return <View style={styles.loadingCircle}><LoadingCircle /></View>;
}

return (
  // regular content
)
```

Reply:
Oh wait, not quite. Since we want to render the title/author/image while content is loading (see above) we don't want to just hide the entire screen if it's loading.

Leave this as one return statement but wrap the video and markdown components in this `loading ? : ` guard